python_modules/dagster/dagster/_core/definitions/asset_daemon_cursor.py

```diff
@@ -296,3 +298,30 @@ def serialize(self) -> str:
             }
         )
         return serialized
+
+
+@whitelist_for_serdes
+class BackcompatAssetDaemonEvaluationInfo(NamedTuple):
+    serialized_cursor: str
+
+    def get_asset_daemon_cursor(self, asset_graph: AssetGraph) -> AssetDaemonCursor:
+        return AssetDaemonCursor.from_serialized(self.serialized_cursor, asset_graph)
+
+    @staticmethod
+    def from_compressed(compressed: str) -> "BackcompatAssetDaemonEvaluationInfo":
+        """This method takes a b64 encoded, zlib compressed string and returns the original
+        BackcompatAssetDaemonEvaluationInfo object.
+        """
+        decoded_bytes = base64.b64decode(compressed)
+        decompressed_bytes = zlib.decompress(decoded_bytes)
+        decoded_str = decompressed_bytes.decode("utf-8")
+        return deserialize_value(decoded_str, BackcompatAssetDaemonEvaluationInfo)
+
+    def to_compressed(self) -> str:
+        """This method compresses the serialized cursor and returns a b64 encoded string to be
+        stored as a string value.
+        """
+        serialized_bytes = serialize_value(self).encode("utf-8")
+        compressed_bytes = zlib.compress(serialized_bytes)
+        encoded_str = base64.b64encode(compressed_bytes)
+        return encoded_str.decode("utf-8")
```

Comment:
you could consider adding these to `serdes` but i dont think thats critical `(de)serialize_gzb64_value` or something

Reply:
yeah I figured I'd wait until there was another usecase for this sort of pattern before "elevating" it as a serdes function.